### PR TITLE
Fixing ONNX Debug mode on Windows, adding ONNX Debug CI for Windows

### LINKED ISSUE #1880: At least one CI build and install ONNX in debug mode
Otherwise, some check may fail silently. Such as https://github.com/onnx/onnx/blob/master/onnx/defs/schema.cc#L820

https://github.com/onnx/onnx/pull/1878 is for fixing the checks. But we need CI to detect such breakage.

### PR DESCRIPTION
**Pull request by vinitra**
To resolve #1880, install onnx from source in CI w/ Debug mode for Windows. Linux / Mac builds are in #2651.

### CODE REVIEW

**Comment by vinitra** [comment]
Looks like the new debug mode CI build passed -- @linkerzhang?

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,11 @@ matrix:
       env: PYTHON_VERSION=python3 ONNX_ML=0
       language: python
       python: "3.6"
+    - os: linux
+      sudo: required
+      env: PYTHON_VERSION=python3 ONNX_ML=0 DEBUG_MODE=1
```
Comment (bddppq) [inline .travis.yml]: Would it be better to directly use `DEBUG=1` (instead of `DEBUG_MODE`) here?

**Comment by linkerzhang** [comment]
> Looks like the new debug mode CI build passed in Travis -- @linkerzhang?

will you also cover it in windows ci?

File: .travis.yml (hunk 1 above)
Comment (wschin) [reply .travis.yml]: `DEBUG` seems a very common environment variable... Maybe `ONNX_DEBUG`?

Comment (vinitra) [reply .travis.yml]: My logic was along the lines of @wschin's comment -- I've renamed it to `ONNX_DEBUG`.

**Comment by vinitra** [comment]
> > Looks like the new debug mode CI build passed in Travis -- @linkerzhang?
> 
> will you also cover it in windows ci?

Added windows build and onnx-ml linux build into Appveyor.

**Comment by vinitra** [comment]
Seeing this error in the Debug build for Windows.
```
gtest.lib(gtest-all.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in test_driver.obj [C:\projects\onnx\.setuptools-cmake-build\onnxifi_test_driver_gtests.vcxproj]

gtest.lib(gtest-all.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in test_driver.obj [C:\projects\onnx\.setuptools-cmake-build\onnxifi_test_driver_gtests.vcxproj]
```
Looks like a discrepancy between debug and release builds in DLLs. Investigating.

**Comment by postrational** [comment]
Looks like there are some merge artifacts on this branch currently.

**Pull request closed without merging**